### PR TITLE
[DATA-663] Add data deletion API

### PR DESCRIPTION
**Pull request by agreenb**
Add gRPC endpoints and requests/responses defined in the ML Data Deletion Scope.

### CODE REVIEW

File: proto/viam/app/data/v1/data.proto [hunk 1]
```diff
@@ -124,3 +133,33 @@ message BinaryMetadata {
...
+message DeleteBinaryDataByFilterRequest {
+  Filter filter = 1;
+}
+
+// DeleteBinaryDataByFilterResponse returns the number of binary files deleted when a filter is provided
+message DeleteBinaryDataByFilterResponse {
+  int64 deleted_count = 1;
+}
+
+// DeleteBinaryDataByIDsRequest deletes the data and metadata of binary (image + file) data by file ids
```
Comment (tahiyasalam) [inline proto/viam/app/data/v1/data.proto:157]: super nit: "... data and metadata of binary (image + file) data _when file ids are provided_"

**Review by AaronCasas — APPROVED** [review]
LGTM!

File: proto/viam/app/data/v1/data.proto (hunk 1 above)
Comment (agreenb) [reply proto/viam/app/data/v1/data.proto]: Thanks, done!

**Comment by agreenb** [comment]
@mcvella The dist/buf tooling is still broken, so locally not getting the js files that CI is generating. Double checked with Adam and he's getting the same result on his machine, so could we force merge this in the interim?

**Pull request merged**